src/main/domain/consent/ConsentFactory.js

```diff
@@ -3,6 +3,8 @@ import {Consent} from './Consent'
 import {ConsentDecoderService} from '../../domain/consent/ConsentDecoderService'
 import {ConsentEncoderService} from '../../domain/consent/ConsentEncoderService'
 
+const emptyVendorPurpose = () => ({consents: {}, legitimateInterests: {}})
```

Comment:
maybe instead of here we could add this as a default in Consent class, as it is with the publisher default object